packages/cli/src/commands/validate.ts

```diff
@@ -122,6 +131,35 @@ export default class Validate extends Command {
     return (trigger as ErrorCondition).error || null
   }
 
+  validateSettings(destination: DestinationDefinition) {
+    const errors: Error[] = []
+    if (destination.mode === 'cloud') {
+      const dest = destination as CloudDestinationDefinition
+      Object.keys(dest.authentication?.fields ?? {}).forEach((field) => {
+        const fieldValues = dest.authentication?.fields[field]
+        //TODO: consider invalidating here -- for now we just warn
+        // this.isInvalid = true
+        const typ = fieldValues?.type
+
+        if (typ === 'boolean' || typ === 'number') {
+          if (!isDefined(fieldValues?.default)) {
+            errors.push(
+              new Error(
+                `The authentication field "${field}" of type "${fieldValues?.type}" does not contain a default value. It is recommended to choose a sane default to avoid validation issues.`
+              )
+            )
+          }
+          if (typeof fieldValues?.default !== typ) {
+            errors.push(
+              new Error(`The default value for field "${field}" is not a ${typ}, but the type is set to ${typ}.`)
```

Comment:
Nit on the language - maybe something like the below? Not a blocker at all.

```suggestion
              new Error(`The default value for field "${field}" is of type "${typeof fieldValues?.default}", but the type is set to "${typ}".`)
```